### DISCUSSION TITLE
Search improvement idea

### DISCUSSION THREAD

**Discussion by vincevargadev** [helix-editor/helix#9676] [Ideas] [answered]
Let's say we have the following entries for the file picker:

```
lib/navigation/utils.js
lib/navigation/routes.js
lib/navigation/navigation.js
lib/navigation/router.js
```

When I type in the file picker "nav nav", I expect it to not show the same results as searching for "nav". For example, I expect `lib/navigation/navigation.js` as the first result as I typed "nav" twice, but in practice, it doesn't get a higher value after I typed "nav nav".

**Comment by sicher** [comment]
It does if you type "navnav". But there seems to be a bug lurking here:

<img width="684" alt="Screenshot 2024-02-20 at 18 25 06" src="https://github.com/helix-editor/helix/assets/65613/e9a3a6a5-e8e4-47c1-a73e-93c25afaac02">

Notice the last entry doesn't match the first "nav". If you continue to filter by typing "nav util" it works as expected, but not with "nav nav"

**Comment by pascalkuthe** [comment, marked as answer]
There is no bug and it's working as intended. Multiple words are treated as `and` so `nav nav` matches `nav` and `nav` which is indeed the same thing as searching just for `nav`. This matches fzf. You need to use `navnav` if you want consequtive matches

**Reply by sicher** [reply]
Ah, that explains it. Thanks!

**Reply by vincevargadev** [reply]
Thank you, the `navnav` helps!

In fact, now I almost never put spaces into my search queries, works much better most of the times.
